src/commands/functions-artifacts-setup-cleanup-policy.ts

```diff
@@ -0,0 +1,191 @@
+  )
+  .withForce("Automatically create or modify cleanup policy")
+  .before(requireAuth)
+  .before(artifactregistry.ensureApiEnabled)
+  .before(requirePermissions, [
+    "artifactregistry.repositories.update",
+    "artifactregistry.versions.delete",
+  ])
+  .action(async (options: any) => {
+    const projectId = needProjectId(options);
+    const location = options.location || "us-central1";
+    const daysToKeep = parseInt(options.days || "5", 10);
+
+    const repoPath = artifacts.makeRepoPath(projectId, location);
+    let repository: artifactregistry.Repository;
+    try {
+      repository = await artifactregistry.getRepository(repoPath);
+    } catch (err: any) {
+      if (err.status === 404) {
+        logBullet(`Repository '${repoPath}' does not exist in Artifact Registry.`);
+        logBullet(
+          `Please deploy your functions first using: ` +
+            `${clc.bold(`firebase deploy --only functions`)}`,
+        );
+        return;
+      }
+      throw err;
+    }
+
+    if (options.none) {
+      if (artifacts.hasCleanupOptOut(repository)) {
+        logBullet(`Repository '${repoPath}' is already opted out from cleanup policies.`);
+        logBullet(`No changes needed.`);
+        return;
+      }
+
+      logBullet(`You are about to opt-out from cleanup policy for repository '${repoPath}'.`);
+      logBullet(
+        `This will prevent suggestions to set up cleanup policy during initialization and deployment.`,
+      );
+
+      const existingPolicy = repository.cleanupPolicies?.[artifacts.CLEANUP_POLICY_ID];
+      if (existingPolicy) {
+        logBullet(`Note: This will remove the existing cleanup policy from the repository.`);
+      }
+
+      const confirmOptOut = await confirm({
+        ...options,
+        default: true,
+        message: "Do you want to continue?",
+      });
+
+      if (!confirmOptOut) {
+        throw new FirebaseError("Command aborted.", { exit: 1 });
+      }
+
+      try {
+        // Remove existing cleanup policy and add opt-out label
+        const policies: artifactregistry.Repository["cleanupPolicies"] = {
+          ...repository.cleanupPolicies,
+        };
+        if (artifacts.CLEANUP_POLICY_ID in policies) {
+          delete policies[artifacts.CLEANUP_POLICY_ID];
+        }
+        const update: Partial<artifactregistry.Repository> = {
+          name: repoPath,
+          labels: { ...repository.labels, [artifacts.OPT_OUT_LABEL_KEY]: "true" },
+          cleanupPolicies: policies,
+        };
+        await artifacts.updateRepository(update);
+        logSuccess(`Successfully opted out from cleanup policy for ${clc.bold(repoPath)}`);
+        return;
+      } catch (err: unknown) {
+        throw new FirebaseError("Failed to opt-out from artifact registry cleanup policy", {
+          original: err as Error,
+        });
+      }
+    }
+
+    if (isNaN(daysToKeep) || daysToKeep <= 0) {
```

Comment:
I wonder if we should allow zero with a fudge factor (e.g. 5m)

Reply:
Like that.